### ISSUE TITLE
Some question regarding seq2seq library

### ISSUE THREAD

**Issue by getao** [farizrahman4u/seq2seq#23]
Hi, thanks for sharing the wonderful library which can greatly help me implement a seq2seq model. However, while using the library, I found some problems.

  1.The examples in  seq2seq package uses mse loss function while in my experiments, I used categorical_crossentropy loss because the output layer is a softmax over vocabulary. To my surprise, the model using categorical_crossentropy performs poorly and the loss is very large. But if I change loss function to mse, as the example codes do, the model performs well. I'm curious why mse loss function performs better than categorical_crossentropy.

2.I found that a 1-layer(depth) seq2seq runs more slowly than a 3-layer(depth) seq2seq model, which is very strange. The time for training a batch in a 1-layer seq2seq is about 2000 seconds while training one batch in a 3-layer seq2seq only needs about 950 seconds. The model is as follows:

print('Build model...')
model = Sequential()
model.add(Embedding(max_features, 100, input_length=maxlen, mask_zero=True,batch_input_shape=(batch_size,maxlen)))
seq2seq = Seq2seq(batch_input_shape=(batch_size,maxlen,100),
        hidden_dim=50,
        output_dim=nb_classes,
        output_length=maxlen,
        depth=(1,1)) #depth=(3,3)
model.add(seq2seq)

Could you help me?

Thanks,
Tao

**Comment by farizrahman4u** [comment]
I am not on my dev machine right now. Are you sure your training data's `Y` is also categorical? It should look like `[0,0,0,0,1,0,0,0]...`. Regarding the depth issue, I will have to check it out.

**Comment by getao** [comment]
Yes, Y is categorical and its shape is as follows:
Y = np.zeros((len(batch_size), maxlen, nb_classes), dtype=np.bool)
It's the exactly the same as the code in https://github.com/nicolas-ivanov/debug_seq2seq. But the code in https://github.com/nicolas-ivanov/debug_seq2seq also uses MSE instead of categorical_crossentropy as loss function, which really confuses me.

Thanks

**Comment by farizrahman4u** [comment]
You should add a  `softmax` activation to the end of  your model to make the output categorical.

**Comment by getao** [comment]
Hi, I added the softmax activation layer and the performance improves. But the time problem still exists. 1-layer seq2seq LSTM runs much more slowly than 3-layer seq2seq LSTM.

**Comment by farizrahman4u** [comment]
Should be fixed now.
